### PR TITLE
dyncom: Fix translation of thumb REVSH

### LINKED ISSUE #1957: Skyeye: Incorrect translation of REVSH (thumb)
Skyeye appears to translate thumb REVSH into ANDEQ. 

TODO: More investigation.

### PR DESCRIPTION
**Pull request by MerryMage**
Closes #1957.

Test included.

### CODE REVIEW

File: src/tests/core/arm/arm_test_common.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+u8 TestEnvironment::TestMemory::Read8(VAddr addr) {
+    auto iter = data.find(addr);
+    if (iter == data.end()) {
+        return addr; // Some arbitrary data
+    }
+    return iter->second;
+}
+
+u16 TestEnvironment::TestMemory::Read16(VAddr addr) {
+    return Read8(addr) | Read8(addr + 1) << 8;
```
Comment (MerryMage) [inline src/tests/core/arm/arm_test_common.cpp]: `<<`s here shift past size of width of type, same below

**Comment by MerryMage** [comment]
OS X build complains about not being able to find `glad` symbols on my system. Not sure exactly why.

File: src/core/arm/dyncom/arm_dyncom_thumb.cpp [hunk 2]
```diff
@@ -293,15 +295,22 @@ ThumbDecodeStatus TranslateThumbInstruction(u32 addr, u32 instr, u32* ainstr, u3
...
                 0xE6BF0F30, // REV
                 0xE6BF0FB0, // REV16
+                0,          // undefined
                 0xE6FF0FB0, // REVSH
             };
 
-            *ainstr = subset[BITS(tinstr, 6, 7)] // base
-                | (BITS(tinstr, 0, 2) << 12)     // Rd
-                | BITS(tinstr, 3, 5);            // Rm
+            size_t subset_index = BITS(tinstr, 6, 7);
```
Comment (wwylele) [inline src/core/arm/dyncom/arm_dyncom_thumb.cpp]: `std::size_t`

**Comment by bunnei** [comment]
What is the deal with the OS X errors? Do you plan on addressing these?

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-07-28T10:52:15Z: https://github.com/citra-emu/citra/compare/c523ff2a98b334b8d2d6d14d2e550f15305f639d...MerryMage:60c93ca3b528ec1d10feb2366a8d0b48c23da271

**Comment by MerryMage** [comment]
@bunnei Leaving out test for now.

**Comment by bunnei** [comment]
Thanks

**Pull request merged**